pkg/scheduler/queue/queue.go

```diff
@@ -389,7 +389,7 @@ func (q *RequestQueue) GetConnectedQuerierWorkersMetric() float64 {
 	return float64(q.connectedQuerierWorkers.Load())
 }
 
-type querierConnection struct {
+type getNextRequestForQuerierCall struct {
```

Comment:
`getNextRequestForQuerierCall` sounds like method name, not name of type to me. I'd suggest to rename it to a noun that describes the type. Verbs are used for methods.

Reply:
some possibilities:
`readyQuerier`
`readyQuerierConnection`/`readyQuerierConn`
`waitingQuerierConnection`/`waitingQuerierConn`

Reply:
I don't love the name either... I wanted to make it clear that this represented a call to the `GetNextRequestForQuerier()` method that is waiting for a query.

I intentionally avoided names like `querier` or `querierConnection` as those are different concepts - a querier has many connections, and a single connection (managed by `Scheduler.QuerierLoop()`) calls `GetNextRequestForQuerier()` many times during its life.

Reply:
I don’t feel strongly enough to hold this up, I think we’ve definitely improved with the naming changes in the PR.

As some of the changes settle we could always rename later if something better makes itself apparent

Reply:
maybe just dropping the `get` is enough non-verbiness for now

Reply:
That would work - might make it `waitingNextRequestForQuerierCall` to make it super clear.